providers/nitro/nitro-helper/src/tracing_layer.rs

```diff
@@ -294,10 +308,13 @@ impl Log {
 
                 let mut value_len_raw = [0; 8];
                 value_len_raw.copy_from_slice(&raw[index_value_len_begin..index_value_len_end]);
-                let value_len = u64::from_le_bytes(value_len_raw) as usize;
+                let value_len_u64 = u64::from_le_bytes(value_len_raw);
+                if value_len_u64 > usize::MAX as u64 {
+                    break;
+                }
 
                 let index_value_begin = index_value_len_end;
-                let index_value_end = index_value_begin + value_len;
+                let index_value_end = index_value_begin + value_len_u64 as usize;
                 flag_index_next = index_value_end + 1;
                 let value_raw = &raw[index_value_begin..index_value_end];
                 log.update(key_raw, value_raw);
```

Comment:
not sure if this may still panic if those indices are invalid or e.g. `index_value_begin + value_len_u64` (`+ 1`) overflows?